packages/jellyfish-api-core/__tests__/category/masternode/updatemasternode.test.ts

```diff
@@ -0,0 +1,576 @@
+    })
+  })
+
+  it('should update multiple address simultaneously with bech32 address', async () => {
+    const initialAddress = await client.wallet.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(initialAddress)
+
+    await container.generate(20)
+
+    const ownerAddress = await client.wallet.getNewAddress()
+    const operatorAddress = await client.wallet.getNewAddress()
+    const rewardAddress = await client.wallet.getNewAddress()
+    await client.masternode.updateMasternode(masternodeId, {
+      ownerAddress,
+      operatorAddress,
+      rewardAddress
+    })
+
+    await container.generate(65)
+
+    const masternodes = await client.masternode.listMasternodes()
+
+    expect(masternodes[masternodeId]).toBeTruthy()
+    expect(masternodes[masternodeId]).toStrictEqual({
+      operatorAuthAddress: operatorAddress,
+      ownerAuthAddress: ownerAddress,
+      rewardAddress: rewardAddress,
+      creationHeight: expect.any(Number),
+      resignHeight: expect.any(Number),
+      resignTx: expect.any(String),
+      collateralTx: expect.any(String),
+      state: 'ENABLED',
+      mintedBlocks: expect.any(Number),
+      ownerIsMine: expect.any(Boolean),
+      localMasternode: expect.any(Boolean),
+      operatorIsMine: expect.any(Boolean),
+      targetMultipliers: expect.any(Object)
+    })
+  })
+
+  it('should update multiple address simultaneously with legacy address', async () => {
+    const initialAddress = await client.wallet.getNewAddress('', AddressType.LEGACY)
+    const masternodeId = await client.masternode.createMasternode(initialAddress)
+
+    await container.generate(20)
+
+    const ownerAddress = await client.wallet.getNewAddress('', AddressType.LEGACY)
+    const operatorAddress = await client.wallet.getNewAddress('', AddressType.LEGACY)
+    const rewardAddress = await client.wallet.getNewAddress('', AddressType.LEGACY)
+    await client.masternode.updateMasternode(masternodeId, {
+      ownerAddress,
+      operatorAddress,
+      rewardAddress
+    })
+
+    await container.generate(65)
+
+    const masternodes = await client.masternode.listMasternodes()
+
+    expect(masternodes[masternodeId]).toBeTruthy()
+    expect(masternodes[masternodeId]).toStrictEqual({
+      operatorAuthAddress: operatorAddress,
+      ownerAuthAddress: ownerAddress,
+      rewardAddress: rewardAddress,
+      creationHeight: expect.any(Number),
+      resignHeight: expect.any(Number),
+      resignTx: expect.any(String),
+      collateralTx: expect.any(String),
+      state: 'ENABLED',
+      mintedBlocks: expect.any(Number),
+      ownerIsMine: expect.any(Boolean),
+      localMasternode: expect.any(Boolean),
+      operatorIsMine: expect.any(Boolean),
+      targetMultipliers: expect.any(Object)
+    })
+  })
+
+  it('should updateMasternode with utxos', async () => {
+    const balance = await container.call('getbalance')
+    expect(balance >= 2).toBeTruthy()
```

Comment:
Kinda weird to write it like this, it should be

```suggestion
    expect(balance).toBeGreaterThanOrEqual(2)
```